47.md

```diff
@@ -63,6 +63,267 @@ Response:
 }
 ```
 
+Below are the commands that are defined in this NIP.
+
+### Pay Invoice
+
+Request:
+```jsonc
+{
+    "method": "pay_invoice",
+    "params": {
+        "invoice": "lnbc50n1..." // invoice to pay, required
+    }
+}
+```
+
+Response:
+```jsonc
+{
+    "result_type": "pay_invoice",
+    "error": {
+        "code": "UNAUTHORIZED",
+        "message": "human readable error message"
+    },
+    "result": {
+        "preimage": "0123456789abcdef..." // preimage of the payment
+    }
+}
+```
+
+### Keysend
+
+Request:
+```jsonc
+{
+    "method": "pay_keysend",
+    "params": {
+        "amount": 123, // invoice amount in msats, required
+        "pubkey": "03...", // payee pubkey, required
+        "message": "some message", // keysend message, optional
+        "preimage": "0123456789abcdef...", // preimage of the payment, optional
+        "tlv_records: [ // tlv records, optional
+            {
+                "type": 5482373484, // tlv type
+                "value": "fajsn341414fq" // hex encoded tlv value
+            }
+        ]
+    }
+}
+```
+
+Response:
+```jsonc
+{
+    "result_type": "pay_keysend",
+    "error": {
+        "code": "UNAUTHORIZED",
+        "message": "human readable error message"
+    },
+    "result": { // result, object. null in case of error.
+        "preimage": "0123456789abcdef...", // preimage of the payment
+        "payment_hash": "31afdf1..", // payment hash of the payment
+    }
+}
+```
+
+### Make Invoice
+
+Request:
+```jsonc
+{
+    "method": "make_invoice",
+    "params": {
+        "amount": 123, // invoice amount in msats, required
+        "description": "some description", // invoice description, optional
+        "description_hash": "31afdf1..", // invoice description hash, optional
+        "preimage": "0123456789abcdef...", // preimage for the payment, optional
+        "expiry": 3600, // invoice expiry in seconds, optional
+    }
+}
+```
+
+Response:
+```jsonc
+{
+    "result_type": "make_invoice",
+    "error": {
+        "code": "UNAUTHORIZED",
+        "message": "human readable error message"
+    },
+    "result": { // result, object. null in case of error.
+        "invoice": "lnbc50n1...",
+        "payment_hash": "31afdf1.."
+    }
+}
+```
+
+### Lookup Invoice
+
+Request:
+```jsonc
+{
+    "method": "lookup_invoice",
+    "params": {
+        "payment_hash": "31afdf1..", // payment hash of the invoice, one of payment_hash or invoice is required
+        "invoice": "lnbc50n1..." // invoice to lookup
+    }
+}
+```
+
+Response:
+```jsonc
+{
+    "result_type": "lookup_invoice",
+    "error": {
+        "code": "UNAUTHORIZED",
+        "message": "human readable error message"
+    },
+    "result": {
+        "invoice": "lnbc50n1...", // the encoded invoice
+        "paid": true // true if the invoice has been paid
```

Comment:
should we return the same information as one entry from the `list_payments` call?
```
{
           "invoice": "string", // encoded invoice
           "payee_pubkey": "string", // pubkey of the payee, used for keysend
           "description": "string",
           "description_hash": "string",
           "preimage": "string",
           "payment_hash": "string",
           "amount": 123, // value in msats
           "fees_paid": 123, // value in msats
           "settled_at": unixtimestamp,
           "metadata": {} // generic metadata that can be used to add things like zap/boostagram details for a payer name/comment/etc.
}
```